contracts/zero-ex/contracts/src/features/interfaces/IERC721OrdersFeature.sol

```diff
@@ -0,0 +1,283 @@
+  you may not use this file except in compliance with the License.
+  You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+  Unless required by applicable law or agreed to in writing, software
+  distributed under the License is distributed on an "AS IS" BASIS,
+  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+  See the License for the specific language governing permissions and
+  limitations under the License.
+
+*/
+
+pragma solidity ^0.6.5;
+pragma experimental ABIEncoderV2;
+
+import "@0x/contracts-erc20/contracts/src/v06/IERC20TokenV06.sol";
+import "../libs/LibERC721Order.sol";
+import "../libs/LibSignature.sol";
+import "../../vendor/IERC721Token.sol";
+
+
+/// @dev Feature for interacting with ERC721 orders.
+interface IERC721OrdersFeature {
+
+    /// @dev Emitted whenever an `ERC721Order` is filled.
+    /// @param direction Whether the order is selling or 
+    ///        buying the ERC721 token. 
+    /// @param erc20Token The address of the ERC20 token.
+    /// @param erc20TokenAmount The amount of ERC20 token 
+    ///        to sell or buy.
+    /// @param erc721Token The address of the ERC721 token.
+    /// @param erc721TokenId The ID of the ERC721 asset.
+    /// @param maker The maker of the order.
+    /// @param taker The taker of the order.
+    /// @param nonce The unique maker nonce in the order.
+    event ERC721OrderFilled(
+        LibERC721Order.TradeDirection direction,
+        IERC20TokenV06 erc20Token,
+        uint256 erc20TokenAmount,
+        IERC721Token erc721Token,
+        uint256 erc721TokenId,
+        address maker,
+        address taker,
+        uint256 nonce
+    );
+
+    /// @dev Emitted whenever two `ERC721Order`s are matched.
+    /// @param erc20Token The ERC20 token of the two orders.
+    ///        If `sellOrder.erc20Token` is the native token
+    ///        and `buyOrder.erc20Token` is the wrapped native
+    ///        token, this will be `NATIVE_TOKEN_ADDRESS`.
+    /// @param erc721Token The address of the ERC721 token.
+    /// @param erc721TokenId The ID of the ERC721 asset.
+    /// @param sellOrderMaker The maker of the sell order.
+    /// @param buyOrderMaker The maker of the buy order.
+    /// @param sellOrderNonce The nonce of the sell order.
+    /// @param buyOrderNonce The nonce of the buy order.
+    /// @param matcher The address that called the `matchERC721Orders`
+    ///        or `batchMatchERC721Orders` function.
+    /// @param profit The amount of profit earned by the matcher,
+    ///        denominated in `erc20Token`.
+    event ERC721OrdersMatched(
+        IERC20TokenV06 erc20Token,
+        IERC721Token erc721Token,
+        uint256 erc20TokenAmount,
+        uint256 erc721TokenId,
+        address sellOrderMaker,
+        address buyOrderMaker,
+        uint256 sellOrderNonce,
+        uint256 buyOrderNonce,
+        address matcher,
+        uint256 profit
```

Comment:
Is this worth logging? Matches are always deterministic so it can be computed off-chain.